loopy/target/c/__init__.py

```diff
@@ -894,13 +894,14 @@ def get_function_declaration(self, codegen_state, codegen_result,
         else:
             name = Value("static void", name)
 
+        idis = self.preprocess_idis(
+            kernel, codegen_result.get_idis_for_subkernel(kernel, subkernel))
+
         return FunctionDeclarationWrapper(
                 FunctionDeclaration(
                     name,
-                    [self.idi_to_cgen_declarator(codegen_state.kernel, idi)
-                     for idi in self.preprocess_idis(
-                         codegen_state.kernel,
-                         codegen_state.implemented_data_info)]
+                    [self.idi_to_cgen_declarator(kernel, idi)
+                     for idi in idis]
```

Comment:
What difference does this make? If it's important, explain in a comment!

Reply:
I'm not sure what you mean - the previous code was incorrect (the arguments to `preprocess_idis`). I restructured this a little to reduce the forests of parentheses ;-)

Reply:
Hm. I see the change now. One thing that we may have to negotiate is that I find the "forest of parentheses" version easier to reason about. (The value is unbound and therefore I can tell that it won't be used/changed below.)